homeassistant/components/ambee/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow to configure the Ambee integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from ambee import Ambee, AmbeeError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+
+class AmbeeFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Ambee."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            try:
+                client = Ambee(
+                    api_key=user_input[CONF_API_KEY],
+                    latitude=user_input[CONF_LATITUDE],
+                    longitude=user_input[CONF_LONGITUDE],
+                    session=session,
+                )
+                await client.air_quality()
+            except AmbeeError:
```

Comment:
Would you be able to distinguish between cannot connect and invalid API key?

Reply:
Not at this moment. I can make a change in the library to handle that and spit out a matching error 👍

Reply:
![image](https://user-images.githubusercontent.com/195327/121272862-495fe300-c8c7-11eb-89b4-d13de5d9242c.png)

Done.